service/frontend/workflowHandler.go

```diff
@@ -908,24 +914,25 @@ func (wh *WorkflowHandler) RespondActivityTaskCompleted(
 		tag.BlobSizeViolationOperation("RespondActivityTaskCompleted"),
 	); err != nil {
 		// result exceeds blob size limit, we would record it as failure
-		failRequest := &gen.RespondActivityTaskFailedRequest{
+		failRequest := &types.RespondActivityTaskFailedRequest{
 			TaskToken: completeRequest.TaskToken,
 			Reason:    common.StringPtr(common.FailureReasonCompleteResultExceedsLimit),
 			Details:   completeRequest.Result[0:sizeLimitError],
 			Identity:  completeRequest.Identity,
 		}
-		err = wh.GetHistoryClient().RespondActivityTaskFailed(ctx, &h.RespondActivityTaskFailedRequest{
+		err = wh.GetHistoryClient().RespondActivityTaskFailed(ctx, &types.HistoryRespondActivityTaskFailedRequest{
```

Comment:
also add `err = thrift.FromError(err)` after this call?

Reply:
Updated.